sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePlugin.scala

```diff
@@ -54,6 +54,9 @@ object ScalaNativePlugin extends AutoPlugin {
     val nativeDump =
       taskKey[Boolean](
         "Shall native toolchain dump intermediate NIR to disk during linking?")
+
+    val nativeTarget =
+      taskKey[String]("The LLVM target triple for cross compiling.")
```

Comment:
Personally, I would go with Option[String] and withTargetTriple(value: String). All `nativeX` settings including nativeTargetTriple exist only for backward compatibility and would be removed in the future. The only remaining way to interact with them would be done via nativeConfig in the same way as it's done in scala.js.

Reply:
I would add that, since `nativeTarget` didn't exist before, there is no backward compatibility to take care of. So we should *only* have it in the `nativeConfig`.

Reply:
I am a little confused. I looked at the Scala.js plugin and that is much more complex so it wasn't much help.

Right now `nativeTarget` is private in the internal plugin. We are moving it to be exposed in the plugin so the user can set it, correct? We can rename to `targetTriple` for the sbt users.

I do think it makes sense to make it an `Option[String]` in the `NativeConfig` as then the downstream processing is safer. If we make it `Option[String]` in the plugin then it looks weird when setting it.

Here is what it looks like added into `sandbox` for testing as an Option.
```scala
lazy val sandbox =
  project
    .in(file("sandbox"))
    .enablePlugins(MyScalaNativePlugin)
    .settings(scalacOptions -= "-Xfatal-warnings")
    .settings(targetTriple := Some("x86_64-apple-macosx10.14.0"))
    .settings(noPublishSettings)
    .dependsOn(nscplugin % "plugin", allCoreLibs, testInterface % Test)
```

Also, shouldn't it be a `settingKey` instead of a `taskKey`? Shouldn't all the "settings" be settings - not much of an sbt plugin expert? An internal task could be used to convert the setting from a `String` to an `Option[String]` maybe?

Reply:
@ekrich I definitely agree with you and I love idea "default" targetTriple.

Maybe call it `nativeTargetTriple`?

Reply:
If I reading correctly, I think the suggestion was just to make it `targetTriple`.

Reply:
@ekrich I'm not sure that it is good idea to remove `native` prefix for only one setting.

Reply:
@catap I think I need to wait for more guidance. I am not exactly sure what @WojciechMazur and @sjrd are saying. See their comments above.

Reply:
Thanks, that looks much better in the code. Down in `GenCode` it gets converted back to `String` to avoid more changes.